### PR TITLE
[US3956] feat(ephermal_disk) adding a quorum property to the dataset.

### PR DESCRIPTION
**Pull request by pawanpraka1**
This property should be inheritable, and
should be allowed to set only once from false to true

Signed-off-by: Pawan <pawanprakash101@gmail.com>

### CODE REVIEW

File: tests/cbtest/script/test_uzfs.sh [hunk 1]
```diff
@@ -192,6 +192,10 @@ run_zvol_tests()
 	log_must $ZFS set sync=standard $src_pool/$src_vol
 	log_must check_prop "$src_pool/$src_vol" sync standard
 
+	log_must check_prop "$src_pool/$src_vol" quorum off
+	log_must $ZFS set quorum=on $src_pool/$src_vol
+	log_must_not $ZFS set quorum=off $src_pool/$src_vol
```
Comment (vishnuitta) [inline tests/cbtest/script/test_uzfs.sh:197]: one more testcase for: `log_must $ZFS set quorum=on $src_pool/$src_vol ` with quorum already being on

File: module/zcommon/zfs_prop.c [hunk 2]
```diff
@@ -350,6 +350,9 @@ zfs_prop_init(void)
 	zprop_register_index(ZFS_PROP_CANMOUNT, "canmount", ZFS_CANMOUNT_ON,
 	    PROP_DEFAULT, ZFS_TYPE_FILESYSTEM, "on | off | noauto",
 	    "CANMOUNT", canmount_table);
+	zprop_register_index(ZFS_PROP_QUORUM, "quorum", 0, PROP_INHERIT,
```
Comment (vishnuitta) [inline module/zcommon/zfs_prop.c:353]: enum should be good to set the quorum like ZFS_QUORUM_OFF, ZFS_QUORUM_ON

**Review by vishnuitta — APPROVED** [review]
changes are good. Some more comments, and can be taken in next PR

File: module/zfs/dsl_dir.c [hunk 3]
```diff
@@ -1664,6 +1664,48 @@ dsl_dir_set_reservation(const char *ddname, zprop_source_t source,
...
+static void
+dsl_dir_set_quorum_sync(void *arg, dmu_tx_t *tx)
+{
+	dsl_dir_set_qr_arg_t *ddsqra = arg;
+	dsl_pool_t *dp = dmu_tx_pool(tx);
+	dsl_dataset_t *ds;
+
+	VERIFY0(dsl_dataset_hold(dp, ddsqra->ddsqra_name, FTAG, &ds));
+
+	dsl_prop_set_sync_impl(ds, zfs_prop_to_name(ZFS_PROP_QUORUM),
```
Comment (vishnuitta) [inline module/zfs/dsl_dir.c:1676]: adding the command to 'zfs history'

**Comment by vishnuitta** [comment]
Will this work for kernel also?

File: module/zcommon/zfs_prop.c (hunk 2 above)
Comment (pawanpraka1) [reply module/zcommon/zfs_prop.c:353]: for boolean value(on/off) we don't need to have separate enum. ZFS uses boolean_table for that purpose.

**Comment by pawanpraka1** [comment]
yes it will work for kernel also.

File: tests/cbtest/script/test_uzfs.sh (hunk 1 above)
Comment (pawanpraka1) [reply tests/cbtest/script/test_uzfs.sh:197]: next PR.

File: module/zfs/dsl_dir.c (hunk 3 above)
Comment (pawanpraka1) [reply module/zfs/dsl_dir.c:1676]: next PR will have this.

**Pull request merged**